fili-core/src/main/java/com/yahoo/bard/webservice/exception/DataExceptionHandler.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.exception;
+
+import com.yahoo.bard.webservice.web.apirequest.DataApiRequest;
+
+import com.fasterxml.jackson.databind.ObjectWriter;
+
+import java.util.Optional;
+
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.UriInfo;
+
+/**
+ * Allows a customer to inject custom code for handling Exceptions in Fili's DataServlet.
+ *
+ * For example, customers may want to return 500's for some custom exceptions, or return
+ * an empty set when dimension rows are not found.
+ */
+public interface DataExceptionHandler {
```

Comment:
Why do we make `DataExceptionHandler`  and `MetadataExceptionHandler` separate?

Reply:
Because of how the two servlets work. The DataServlet uses asynchronous responses, so we need to pass in the asyncResponse for this method to resume.

The metadata servlets meanwhile use synchronous responses (if you look at their `get*` methods, you'll see  that they return a `Response`), so we need to do the same thing in the Metadata interface.

Kinda hard to put both of those behaviors into one interface.